src/bp/core/services/ghost/memory-cache.test.ts

```diff
@@ -0,0 +1,293 @@
+import 'bluebird-global'
+import 'reflect-metadata'
+
+import LRU from 'lru-cache'
+import sizeof from 'object-sizeof'
+
+import { createSpyObject, MockObject } from '../../misc/utils'
+
+import { ObjectCache } from 'common/object-cache'
+import MemoryObjectCache, { EVENTS } from './memory-cache'
+import { CacheInvalidators } from './cache-invalidators'
+
+const AN_INVALID_KEY = 'invalid key'
+const A_KEY = 'key'
+const ANOTHER_KEY = 'another key'
+const A_VALUE = 'a value'
+const AN_OBJECT_VALUE = { key: 'value' }
+const HUGE_OBJECT_VALUE = Array.from({ length: 40 }, () => Math.floor(Math.random() * 40).toString())
+const MAX_CACHE_SIZE = 24 // MAX_CACHE_SIZE must be less than the size of HUGE_OBJECT_VALUE
+const SYNC_MESSAGE = 'a message'
+
+describe('Memory Cache', () => {
+  let cache: ObjectCache
+  let cacheInvalidator: MockObject<CacheInvalidators.FileChangedInvalidator>
+  let assertCacheIsEmpty: () => void
+
+  beforeEach(() => {
+    const defaultCacheValue = process.core_env.BP_MAX_MEMORY_CACHE_SIZE
+    ;(process.core_env.BP_MAX_MEMORY_CACHE_SIZE as string) = `${MAX_CACHE_SIZE}b`
+
+    cacheInvalidator = createSpyObject<CacheInvalidators.FileChangedInvalidator>()
+    cache = new MemoryObjectCache(cacheInvalidator.T)
+    ;(process.core_env.BP_MAX_MEMORY_CACHE_SIZE as string | undefined) = defaultCacheValue
```

Comment:
You can set `process.env.BP_MAX_MEMORY_CACHE_SIZE = defaultCacheValue ` to avoid all those weird typings and ;( , it will be much more clean